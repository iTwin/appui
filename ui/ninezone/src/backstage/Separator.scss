/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
@import "_variables";

.nz-backstage-separator {
  height: $nz-backstage-divider-height;
  background-color: $nz-backstage-divider-color;
}
