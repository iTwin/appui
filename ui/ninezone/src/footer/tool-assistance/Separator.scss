@import "@bentley/ui-core/lib/colors";

.nz-footer-toolAssistance-separator {
  margin-bottom: 10px;

  > .nz-separator {
    height: 2px;
    width: 100%;
    background-color: $background-4;
  }
}
