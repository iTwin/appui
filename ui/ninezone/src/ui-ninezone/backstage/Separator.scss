/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
@import "_variables";

.nz-backstage-separator {
  height: $nz-backstage-divider-height;
  background-color: $nz-backstage-divider-color;
}
