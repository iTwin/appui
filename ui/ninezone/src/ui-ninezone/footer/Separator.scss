/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
@import "@bentley/ui-core/lib/ui-core/style/themecolors";
@import "_variables";

.nz-footer-separator {
  width: 1px;
  filter: drop-shadow(1px 0 $buic-foreground-body-reverse);
  margin: $nz-footer-indicator-vertical-padding 0;
  background-color: $buic-foreground-disabled;
}
