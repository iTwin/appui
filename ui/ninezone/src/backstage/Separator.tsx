/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
/** @module Backstage */

import * as classnames from "classnames";
import * as React from "react";
import Props from "../utilities/Props";
import "./Separator.scss";

/** Item separator in the [[Backstage]]. */
export default class BackstageSeparator extends React.Component<Props> {
  public render() {
    const className = classnames(
      "nz-backstage-separator",
      this.props.className);

    return (
      <div className={className} style={this.props.style} />
    );
  }
}
