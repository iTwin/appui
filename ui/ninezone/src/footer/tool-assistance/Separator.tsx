/*---------------------------------------------------------------------------------------------
* Copyright (c) 2018 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
/** @module ToolAssistance */

import * as classnames from "classnames";
import * as React from "react";
import CommonProps from "../../utilities/Props";
import "./Separator.scss";

/** Properties of [[ToolAssistanceSeparator]] component. */
export interface ToolAssistanceSeparatorProps extends CommonProps {
  /** Label of separator. */
  label?: string;
}

/** Assistance item separator. Used in [[ToolAssistance]] component. */
export default class ToolAssistanceSeparator extends React.Component<ToolAssistanceSeparatorProps> {
  public render() {
    const className = classnames(
      "nz-footer-toolAssistance-separator",
      this.props.className);

    return (
      <div
        className={className}
        style={this.props.style}
      >
        <div className="nz-label">
          {this.props.label}
        </div>
        <div className="nz-separator" />
      </div>
    );
  }
}
