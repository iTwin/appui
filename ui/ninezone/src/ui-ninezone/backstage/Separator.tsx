/*---------------------------------------------------------------------------------------------
* Copyright (c) 2019 Bentley Systems, Incorporated. All rights reserved.
* Licensed under the MIT License. See LICENSE.md in the project root for license terms.
*--------------------------------------------------------------------------------------------*/
/** @module Backstage */

import * as classnames from "classnames";
import * as React from "react";
import { CommonProps } from "@bentley/ui-core";
import "./Separator.scss";

/** Item separator in the [[Backstage]] component.
 * @beta
 */
export class BackstageSeparator extends React.PureComponent<CommonProps> {
  public render() {
    const className = classnames(
      "nz-backstage-separator",
      this.props.className);

    return (
      <div className={className} style={this.props.style} />
    );
  }
}
