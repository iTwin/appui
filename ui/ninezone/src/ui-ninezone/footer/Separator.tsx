/*---------------------------------------------------------------------------------------------
* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
* See LICENSE.md in the project root for license terms and full copyright notice.
*--------------------------------------------------------------------------------------------*/
/** @packageDocumentation
 * @module Footer
 */

import * as classnames from "classnames";
import * as React from "react";
import { CommonProps, NoChildrenProps } from "@bentley/ui-core";
import "./Separator.scss";

/** Properties of [[FooterSeparator]] component.
 * @beta
 */
export interface FooterSeparatorProps extends CommonProps, NoChildrenProps {
}

/** Footer indicator separator used in [[Footer]] component.
 * @beta
 */
export class FooterSeparator extends React.PureComponent<FooterSeparatorProps> {
  public render() {
    const className = classnames(
      "nz-footer-separator",
      this.props.className);

    return (
      <div
        className={className}
        style={this.props.style}
      />
    );
  }
}
