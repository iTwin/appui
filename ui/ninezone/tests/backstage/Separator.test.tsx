/*---------------------------------------------------------------------------------------------
|  $Copyright: (c) 2018 Bentley Systems, Incorporated. All rights reserved. $
 *--------------------------------------------------------------------------------------------*/
import { mount, shallow } from "enzyme";
import * as React from "react";

import Separator from "@src/backstage/Separator";

describe("<Separator />", () => {
  it("should render", () => {
    mount(<Separator />);
  });

  it("renders correctly", () => {
    shallow(<Separator />).should.matchSnapshot();
  });
});
