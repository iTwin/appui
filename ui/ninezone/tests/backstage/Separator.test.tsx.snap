// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Separator /> renders correctly 1`] = `
<div
  className="nz-backstage-separator"
/>
`;
